Rakefile

```
@@ -11,49 +11,74 @@ namespace :book do
   if version_string.empty?
     version_string = '0'
   end
-  date_string = Time.now.strftime("%Y-%m-%d")
+  date_string = Time.now.strftime('%Y-%m-%d')
   params = "--attribute revnumber='#{version_string}' --attribute revdate='#{date_string}'"
+  header_hash = `git rev-parse --short HEAD`.strip
+
+  # Check contributors list
+  # This checks commit hash stored in the header of list against current HEAD
+  def check_contrib
+    if File.exist?('book/contributors.txt')
+      current_head_hash = `git rev-parse --short HEAD`.strip
+      header = `head -n 1 book/contributors.txt`.strip
+      # Match regex, then coerce resulting array to string by join
+      header_hash = header.scan(/[a-f0-9]{7,}/).join
+
+      if header_hash == current_head_hash
+        puts "Hash on header of contributors list (#{header_hash}) matches the current HEAD (#{current_head_hash})"
+      else
+        puts "Hash on header of contributors list (#{header_hash}) does not match the current HEAD (#{current_head_hash}), refreshing"
+        `rm book/contributors.txt`
+        # Reenable and invoke task again
+        Rake::Task['book/contributors.txt'].reenable
+        Rake::Task['book/contributors.txt'].invoke
+      end
+    end
+  end
```

```suggestion
  # Check contributors list
  # This checks the commit hash stored in the header of the list against the current HEAD
  def check_contrib
    if File.exist?('book/contributors.txt')
      current_head_hash = `git rev-parse --short HEAD`.strip
      header = `head -n 1 book/contributors.txt`.strip
      # Match regex, then coerce resulting array to string by join
      header_hash = header.scan(/[a-f0-9]{7,}/).join

      if header_hash == current_head_hash
        puts "Hash on header of contributors list (#{header_hash}) matches the current HEAD (#{current_head_hash})"
      else
        puts "Hash on header of contributors list (#{header_hash}) does not match the current HEAD (#{current_head_hash}), refreshing"
        `rm book/contributors.txt`
        # Re-enable and invoke task again
        Rake::Task['book/contributors.txt'].reenable
        Rake::Task['book/contributors.txt'].invoke
      end
    end
  end
```

Some small comment fixes.